cmake/Modules/FindQhull.cmake

```diff
@@ -61,34 +109,35 @@ endif()
 
 get_filename_component(QHULL_LIBRARY_DEBUG_NAME "${QHULL_LIBRARY_DEBUG}" NAME)
 
-if(QHULL_INCLUDE_DIR AND QHULL_LIBRARY)
-
-  # Include directories
-  set(QHULL_INCLUDE_DIRS ${QHULL_INCLUDE_DIR})
-  unset(QHULL_INCLUDE_DIR)
-  mark_as_advanced(QHULL_INCLUDE_DIRS)
-
-  # Libraries
-  set(QHULL_LIBRARIES optimized ${QHULL_LIBRARY} debug ${QHULL_LIBRARY_DEBUG})
-  unset(QHULL_LIBRARY)
-  unset(QHULL_LIBRARY_DEBUG)
-  mark_as_advanced(QHULL_LIBRARIES)
-
-endif()
-
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Qhull
   FOUND_VAR QHULL_FOUND
-  REQUIRED_VARS QHULL_LIBRARIES QHULL_INCLUDE_DIRS
+  REQUIRED_VARS QHULL_LIBRARY QHULL_INCLUDE_DIR
 )
 
 if(QHULL_FOUND)
   set(HAVE_QHULL ON)
-  if(NOT QHULL_USE_STATIC)
-    add_definitions("-Dqh_QHpointer")
-    if(MSVC)
-      add_definitions("-Dqh_QHpointer_dllimport")
+  add_library(QHULL::QHULL ${QHULL_LIBRARY_TYPE} IMPORTED)
+  set_target_properties(QHULL::QHULL PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${QHULL_INCLUDE_DIR}")
+  set_property(TARGET QHULL::QHULL APPEND PROPERTY IMPORTED_CONFIGURATIONS "RELEASE")
+  set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES_RELEASE "CXX")
+  set_target_properties(QHULL::QHULL PROPERTIES COMPILE_DEFINITIONS "-Dqh_QHpointer")
+  if(MSVC)
+    set_target_properties(QHULL::QHULL PROPERTIES COMPILE_DEFINITIONS "--Dqh_QHpointer_dllimport")
+  endif()
+
+  if(WIN32 AND NOT QHULL_USE_STATIC)
+    set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_IMPLIB_RELEASE "${QHULL_LIBRARY}")
+  else()
+    set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_LOCATION_RELEASE "${QHULL_LIBRARY}")
+  endif()
+  if(QHULL_LIBRARY_DEBUG)
+    set_property(TARGET QHULL::QHULL APPEND PROPERTY IMPORTED_CONFIGURATIONS "DEBUG")
+    set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES_DEBUG "CXX")
+    if(WIN32 AND NOT QHULL_USE_STATIC)
+      set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_IMPLIB_DEBUG "${QHULL_LIBRARY_DEBUG}")
+    else()
+      set_target_properties(QHULL::QHULL PROPERTIES IMPORTED_LOCATION_DEBUG "${QHULL_LIBRARY_DEBUG}")
     endif()
   endif()
-  message(STATUS "QHULL found (include: ${QHULL_INCLUDE_DIRS}, lib: ${QHULL_LIBRARIES})")
```

Comment:
is it not worth to report this?

Reply:
It's absolutely worth keeping it: I often need to know where my library has been found and it's a simple way to diagnose errors when users share logs.

Reply:
I've removed it because I thought it was a duplicate.

When I configure cloud_viewer (with this PR) I've got this

```
root@367c56bb4824:/opt/pcl/tutorials/cloud_viewer# cmake ../../doc/tutorials/content/sources/cloud_viewer/ -DCMAKE_BUILD_TYPE=Release
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Checking for module 'eigen3'
--   Found eigen3, version 3.2.92
-- Found Eigen: /usr/include/eigen3 (Required is at least version "3.1") 
-- Eigen found (include: /usr/include/eigen3, version: 3.2.92)
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   system
--   filesystem
--   date_time
--   iostreams
--   serialization
--   regex
-- Checking for module 'flann'
--   Found flann, version 1.8.4
-- Found FLANN: /usr/lib/x86_64-linux-gnu/libflann_cpp.so  

...

-- Found Qhull: /usr/lib/x86_64-linux-gnu/libqhull.so  <-----
``` 

I don't know where this message comes from, maybe from one of the cmake find function ?

Anyway if you think the previous message is still needed, I can fix it.

Reply:
For some reason I don't get the *Found Qhull* at every rebuild, but only sometimes.

Reply:
Yep, seems to be printed only at first configure ?

Reply:
Yeah, while the other ones are systematically printed. I don't know why there is a difference.

Reply:
I've re-added the previous message function.